expan/core/experiment.py

```diff
@@ -287,11 +287,55 @@ def _quantile_filtering(self, data, kpis, percentile, threshold_type):
         :return: boolean values indicating whether the row should be filtered
         :rtype: pd.Series
         """
-        method_table = {'upper': lambda x: x > threshold, 'lower': lambda x: x <= threshold}
-        na_replacement={'upper': -float_info.max        , 'lower':  float_info.max}
-
+        method_table   = {'upper': lambda x: x > threshold, 'lower': lambda x: x <= threshold}
+        na_replacement = {'upper': -float_info.max,         'lower':  float_info.max}
 
         for column in data[kpis].columns:
             threshold = np.percentile(data[column].fillna(na_replacement[threshold_type]), percentile)
             flags = flags | data[column].apply(method_table[threshold_type])
         return flags
+
+
+    def chi_square_test_result_and_statistics(self, variant_column, weights, min_counts=5, alpha=0.05):
+        """ Tests the consistency of variant split with the hypothesized distribution.
+        
+        :param variant_column: variant column from the input data frame
+        :param weights: dict with variant names as keys, weights as values
+                        ({<variant_name>:<weight>, ...}
+        :param min_counts: minimum number of observed and expected frequencies (should be at least 5), see 
+                            http://docs.scipy.org/doc/scipy-0.16.1/reference/generated/scipy.stats.chisquare.html
+        :param alpha: significance level, 0.05 by default
+        :return: True(if split is consistent with the given split) or
+                 False(if split is not consistent with the given split)
+        :rtype:  Boolean, float, float
+        """
+        if not variant_column:
+            raise ValueError("Variant split check was cancelled since input variant column is empty or doesn't exist.")
+        if not weights:
+            raise ValueError("Variant split check was cancelled since input weights are empty or doesn't exist.")
+        if len(weights) <= 1 or len(variant_column) <= 1:
+            raise ValueError("Variant split check was cancelled since input weights or the number if categories "
+                             "is less than 2.")
+
+        # Count number of observations per each variant
+        variant_column = pd.Series(variant_column).dropna(axis=0)
+        variants_counts = variant_column.value_counts()
+        observed_counts = pd.DataFrame(variants_counts, columns=['observed'])
+        # Ensure at least a frequency of 5 at every location in observed_counts, otherwise drop the category
+        # see http://docs.scipy.org/doc/scipy-0.16.1/reference/generated/scipy.stats.chisquare.html
+        observed_freqs = observed_counts[observed_counts >= min_counts].dropna(axis=0)
```

Comment:
`observed_counts` is a `DataFrame`, not a `Series`, so I don't like `observed_counts >= min_counts`. It could be `observed_counts['observed'] >= min_counts`

Or perhaps `.query` is better, I only recently discovered it in pandas:


    observed_freqs = observed_counts.query('observed >= min_counts')

If that cannot "see" `min_counts`, then perhaps try this instead:

    observed_freqs = observed_counts.query('observed >= {}'.format(min_counts))

Reply:
never used .query, looks nice)